### PR TITLE
feat: remote_write_receiver option

### PR DESCRIPTION
**Pull request by Guillaumebeuzeboc**
## Issue
the juju cos setup cannot receive data from the grafana-agent if the grafana-agent is not running inside juju.

## Solution
Add a charm option to enable the remote `remote_write_receiver` feature even when no relation is established.

## Context
IoT devices cannot run juju due to resources constraint, so they must run grafana-agent snap standalone.
This way the COS stack can be used on servers and snaps can be used on devices to export data.

## Notes

This is a draft PR as a suggestion as I am exploring the usage of COS for robots.
The PR is based on my recent learning of Juju/Charm and COS.

### CODE REVIEW

**Comment by dstathis** [comment]
This seems like a good idea. What do you think @simskij?

**Review by simskij — APPROVED** [review]

File: config.yaml [hunk 1]
```diff
@@ -72,3 +72,8 @@ options:
       automatically deduced from it).
       See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
     type: string
+  remote_write_receiver:
```
Comment (sed-i) [inline config.yaml:75]: I think we shouldn't have this config option. Instead, always enable remote write:
Currently, when we enable remote-write, it's enabled for everyone; afaict, there isn't greater risk in enabling it to begin with.

**Review by sed-i — CHANGES_REQUESTED** [review]
Thanks @Guillaumebeuzeboc!
Ideally we'd want to keep config options minimal or scalable.
Since juju config options cannot be nested, we try to stick to minimal.

File: src/charm.py [hunk 2]
```diff
@@ -667,7 +667,9 @@ def _generate_command(self) -> str:
         args.append(f"--web.external-url={external_url}")
         args.append("--web.route-prefix=/")
 
-        if self.model.relations[DEFAULT_REMOTE_WRITE_RELATION_NAME]:
+        if self.model.relations[DEFAULT_REMOTE_WRITE_RELATION_NAME] or config.get(
+            "remote_write_receiver"
+        ):
             args.append("--web.enable-remote-write-receiver")
```
Comment (simskij) [inline src/charm.py:673]:
```suggestion
        args.append("--web.enable-remote-write-receiver")
```

File: config.yaml [hunk 3]
```diff
@@ -72,3 +72,8 @@ options:
...
+    description: |
+      enforce the remote-write-receiver feature
+    type: boolean
+    default: false
```
Comment (simskij) [inline config.yaml:79]:
```suggestion
```

**Review by simskij — APPROVED** [review]

**Comment by simskij** [comment]
@Guillaumebeuzeboc would you mind squashing your commits into a signed commit to allow for merging? Thanks.

**Comment by Guillaumebeuzeboc** [comment]
> @Guillaumebeuzeboc would you mind squashing your commits into a signed commit to allow for merging? Thanks.

Done, let me know if it's all good.
Thanks!

**Pull request merged**